.github/workflows/ci_nightly.yml

```
@@ -51,3 +51,20 @@ jobs:
           device: ${{ matrix.device.soc }}
           target: ${{ matrix.device.target }}
           toolchain: nightly
+
+    create-issue-on-failure:
+    name: Create Issue on Failure
+    runs-on: ubuntu-latest
+    needs: esp-hal-nightly
+    if: failure()
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4  # Ensures the repo is cloned before accessing the issue template
+
+      - name: Create GitHub Issue
+        uses: JasonEtco/create-an-issue@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          filename: .github/ISSUE_TEMPLATE/nightly_ci.md
+          update_existing: true
```

```suggestion
      - if: failure()
        name: Create GitHub Issue
        uses: JasonEtco/create-an-issue@v2
        with:
          filename: .github/ISSUE_TEMPLATE/nightly_ci.md
          update_existing: true
```

I think this should work?